python/ray/experimental/client/server/core_ray_api.py

```diff
@@ -39,3 +40,34 @@ def close(self, *args, **kwargs):
     # doesn't currently support them.
     def __getattr__(self, key: str):
         return getattr(ray, key)
+
+
+class CoreRayServerAPI(CoreRayAPI):
```

Comment:
Why does this class need to be separate from the server impl? Consider combining them into one class.

Reply:
They do represent different things; one is the implementation of the API from the perspective of the client-server and one is from the perspective of the in-cluster ray worker executing these things. That insight helped make the whole thing easier to digest and separate out the cases for, without having large if-else blocks

Reply:
Now, that's also probably worth a pydoc ;)